src/hangar/metadata.py

```diff
@@ -72,7 +72,15 @@ def __len__(self):
         int
             number of metadata key/value pairs.
         '''
-        return len(self._Query.metadata_names())
+        if not self._is_conman:
+            self._dataTxn = TxnRegister().begin_reader_txn(self._dataenv)
+        try:
+            metadataCountKey = parsing.metadata_count_db_key()
+            metadata_count = self._dataTxn.get(metadataCountKey)
+        finally:
+            if not self._is_conman:
+                self._dataTxn = TxnRegister().abort_reader_txn(self._dataenv)
+        return int(metadata_count) if metadata_count is not None else 0
```

Comment:
lmdb transactions have the ability to set a default value if the lookup fails. This ternary check is unnecessary, and could have been eliminated with:

```
metaCountVal = self._dataTxn.get(metaCountKey, default='0'.encode())
```